master/buildbot/config.py

```diff
@@ -688,10 +689,25 @@ def __init__(self, name=None, slavename=None, slavenames=None,
         self.slavebuilddir = slavebuilddir
 
         # remainder are optional
-        if category is not None and not isinstance(category, str):
-            error("builder '%s': category must be a string" % (name,))
+        if category and tags:
+            error("builder '%s': category is being replaced by tags; you should only specify tags" % (name,))
+
+        if category:
+            if not isinstance(category, str):
+                error("builder '%s': category must be a string" % (name,))
+            tags = [category]
+
+            self.category = category  # Set this for legacy reasons
```

Comment:
This line was not part of the Nine review. I added it just in case subclasses had come to depend on it being there, but maybe that's not a concern.